frontend/src/components/student/BadgesPage/BadgesPage.js

```diff
@@ -72,6 +85,49 @@ function BadgesPage(props) {
     [rankInfo]
   )
 
+  const badgeContent = useCallback(
+    (badge, index) => {
+      const badgeUnlockedDate = unlockedBadgesList.find((b) => {
+        return b.badge.title === badge.title
+      })?.unlockedDateMillis
+
+      const tooltipText = !!badgeUnlockedDate
+        ? `Odblokowano: ${moment(badgeUnlockedDate).format('DD.MM.YYYY, HH:mm')}`
+        : 'Odznaka nadal nie została zdobyta.'
+
+      return (
+        <Col md={4} className={'text-center d-flex flex-column align-items-center'} key={index + Date.now()}>
+          <img
+            data-for={'badge-' + index}
+            data-tip={tooltipText}
+            style={{ opacity: !!badgeUnlockedDate ? 1 : 0.4 }}
+            width={100}
+            src={base64Header + badge.file.file}
+            alt={'badge-icon'}
+          />
+
+          <p className={'m-0'}>
+            <strong style={{ opacity: !!badgeUnlockedDate ? 1 : 0.4 }}>{badge.title}</strong>
+          </p>
+          <p style={{ opacity: !!badgeUnlockedDate ? 1 : 0.4 }} className={'px-2'}>
+            {badge.description}
+          </p>
+
+          <ReactTooltip
+            id={'badge-' + index}
+            place='top'
+            type='dark'
+            effect='solid'
+            multiline
+            event='mouseover mouseenter'
+            eventOff='mouseleave mouseout scroll mousewheel blur'
+          />
+        </Col>
+      )
+    },
+    [unlockedBadgesList]
+  )
+
   return (
     <Content>
       {rankInfo === undefined ? (
```

Comment:
Jeśli będziesz na zmianę klikał między odznaki/inny ekran na nawigacji, może ci się zdarzyć że `rankData` wczyta się przed `unlockedBadgesList` i dostaniemy biały ekran

Najprostsze rozwiązanie to zamienić tą linię na:
```suggestion
      rankInfo === undefined || unlockedBadgesList === undefined
```

Możliwe że da się też przenieść check niżej, ale myślę że i tak lepiej wczytać dopiero jak na 100% mamy wszystko wczytane - wydaje mi się też że nie ma potrzeby uwzględniać `lastUnlockedBadge` bo jest na tym samym ładowaniu i jest obsłużone niżej co jeśli jest falsy

Reply:
Po pierwsze nie jestem w stanie zreprodukować błędu, o którym piszesz.

Po drugie nawet jeżeli gdzieś byłby problem to zasugerowana przez ciebie zmiana nie ma prawa naprawić tego błędu, dlatego że jak przyjrzysz się dokładnie to `unlockedBadgesList` nie jest używany nigdzie w fragmencie kodu, którego dotyczy sprawdzenie `undefined`. Wszelkie użycia są sprawdzane wewnątrz `Row`, który jest dosłownie zaraz po `}` zamykającym nawias otwierający w linii 133. Poza tym każde użycie `unlockedBadgesList` jest sprawdzane w kontekście wystąpienia `undefined`. Więc jeżeli faktycznie ten błąd występuje to poproszę o nagranie.

Reply:
Mój błąd, przez to że ten błąd nie był pewny do odtworzenia akurat tak mi się złożyło że zniknął po tej zmianie lokalnie - ale ten refaktor który wprowadziłeś z tego co widzę to naprawił na pewno

Podrzucam nagranie z tego buga zrobione commit wcześniej przed ostatnim
https://user-images.githubusercontent.com/56168780/197021115-4305d0e6-084e-473d-ab2b-0e46acce8589.mp4